packages/discovery-provider/src/tasks/entity_manager/entities/event.py

```diff
@@ -0,0 +1,113 @@
+from src.exceptions import IndexingValidationError
+from src.models.events.event import Event
+from src.tasks.entity_manager.utils import (
+    EntityType,
+    ManageEntityParameters,
+    copy_record,
+    validate_signer,
+)
+from src.utils.structured_logger import StructuredLogger
+
+logger = StructuredLogger(__name__)
+
+
+def validate_create_event_tx(params: ManageEntityParameters):
+    validate_signer(params)
+    metadata = params.metadata
+
+    required_fields = ["event_type"]
+    for field in required_fields:
+        if field not in metadata:
+            raise IndexingValidationError(f"Missing required field: {field}")
+
+
+def create_event(params: ManageEntityParameters):
+    validate_create_event_tx(params)
+
+    event_id = params.entity_id
+    event_record = Event(
+        event_id=event_id,
+        event_type=params.metadata["event_type"],
+        user_id=params.user_id,
+        entity_type=params.metadata["entity_type"],
+        entity_id=params.metadata["entity_id"],
+        end_date=params.metadata["end_date"],
+        is_deleted=params.metadata["is_deleted"],
+        created_at=params.block_datetime,
+        updated_at=params.block_datetime,
+        txhash=params.txhash,
+        blockhash=params.event_blockhash,
+        blocknumber=params.block_number,
+    )
+
+    params.add_record(event_id, event_record)
```

Comment:
Similar to below, we likely don't need to use add_record here, since none of the logic its doing is applicable to this entity type. `params.session.add` is probably fine.